simpleviewinc/keg-components
# Zen 294 modal

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-294)

## Context

* the tap-events-force repo already has a modal component
* might need a little updating, but it's already cross platform

## Goal

* Create a generic modal component with correct styling on web and native platforms
* Allow caller to pass in any components which define the contents of the modal

## Updates

* *src/components/modal/index.js*
  * Added export
* *src/components/modal/modal.js*
  * New modal component with two defined props:
    * visible - determines if modal shows up or not
    * touchBackdrop - callback for when the space outside the modal (backdrop) is clicked or touched (defaults to noop function)
* *src/components/modal/modal.stories.js*
  * Add 2 new modal storyies; include text and button as children (button helps differentiate between clicking inside modal and outside it)
* *src/theme/components/components.js*
  * Export modal component
* *src/theme/components/modal.js*
  * Modal styling for web sizes
  * Waiting to style mobile until getStyles call inside useLayoutEffect is resolved (see attached .js file on issue for mobile styles)
* *src/theme/defaults.json*
  * Add default modal width
* *src/utils/helpers/index.js*
  * Export new noop function util
* *src/utils/helpers/noop.js*
  * New empty function that does nothing. Useful for passing as default function in props so you don't have to check if function is defined inside component

## Testing

* Run storybook for keg-components: yarn sb
* Navigate to Modal > standard
  * Click anywhere outside modal - should get "Touched outside of modal" action
  * Click anywhere inside modal - should not get action
  * Click button - should get "clicked button" action
  * Open inspector and change sizes
    * Style should work for all sizes (no text cutoff, etc)
    * Should get no errors in console
* Navigate to Modal > style overrride
  * Modal should have a yellow background and be smaller
  * Test same functionality as standard modal
